client/state/data-layer/wpcom/comments/index.js

```diff
@@ -156,6 +160,35 @@ export const announceFailure = ( { dispatch, getState }, { siteId, postId } ) =>
 	dispatch( errorNotice( error ) );
 };
 
+export const deleteComment = ( { dispatch }, action ) => {
+	const { siteId, commentId } = action;
+	dispatch(
+		http( {
+			method: 'POST',
+			apiVersion: '1.1',
+			path: `/sites/${ siteId }/comments/${ commentId }/delete`,
+			onSuccess: {
+				...action,
+				type: COMMENTS_DELETE_SUCCESS,
+			},
+			onFailure: {
+				...action,
+				type: COMMENTS_DELETE_FAILURE,
+			},
+		} )
+	);
+};
+
+export const removeCommentFromState = ( { dispatch }, action ) => {
+	const { siteId, postId, commentId } = action;
+	dispatch( removeComment( siteId, postId, commentId ) );
+};
+
+export const announceDeleteFailure = ( { dispatch } ) => {
+	dispatch( errorNotice( translate( 'Could not delete the comment' ) ) );
+};
```

Comment:
This would also be a good place to add the comment back into Calypso on the failure. If we need to we could pass along the original comment in the `onFailure` action (or the original action)

Also, this is a reason for having a proper data model…

```js
announceFailureAndRestoreComment = ( { dispatch }, action ) => {
	dispatch( errorNotice( … ) );

	dispatch( local( addComment( … ) ) );
}
```

local comes from `state/data-layer/wpcom-http/utils` I think